code/__HELPERS/_lists.dm

```diff
@@ -1,827 +1,827 @@
-		return
-	if(fromIndex > toIndex)
-		++fromIndex	//since a null will be inserted before fromIndex, the index needs to be nudged right by one
-
-	L.Insert(toIndex, null)
-	L.Swap(fromIndex, toIndex)
-	L.Cut(fromIndex, fromIndex + 1)
-
-
-//Move elements [fromIndex,fromIndex+len) to [toIndex-len, toIndex)
-//Same as moveElement but for ranges of elements
-//This will preserve associations ~Carnie
-/proc/moveRange(list/L, fromIndex, toIndex, len = 1)
-	var/distance = abs(toIndex - fromIndex)
-	if(len >= distance)	//there are more elements to be moved than the distance to be moved. Therefore the same result can be achieved (with fewer operations) by moving elements between where we are and where we are going. The result being, our range we are moving is shifted left or right by dist elements
-		if(fromIndex <= toIndex)
-			return	//no need to move
-		fromIndex += len	//we want to shift left instead of right
-
-		for(var/i = 0, i < distance, ++i)
-			L.Insert(fromIndex, null)
-			L.Swap(fromIndex, toIndex)
-			L.Cut(toIndex, toIndex + 1)
-	else
-		if(fromIndex > toIndex)
-			fromIndex += len
-
-		for(var/i = 0, i < len, ++i)
-			L.Insert(toIndex, null)
-			L.Swap(fromIndex, toIndex)
-			L.Cut(fromIndex, fromIndex + 1)
-
-//Move elements from [fromIndex, fromIndex+len) to [toIndex, toIndex+len)
-//Move any elements being overwritten by the move to the now-empty elements, preserving order
-//Note: if the two ranges overlap, only the destination order will be preserved fully, since some elements will be within both ranges ~Carnie
-/proc/swapRange(list/L, fromIndex, toIndex, len = 1)
-	var/distance = abs(toIndex - fromIndex)
-	if(len > distance)	//there is an overlap, therefore swapping each element will require more swaps than inserting new elements
-		if(fromIndex < toIndex)
-			toIndex += len
-		else
-			fromIndex += len
-
-		for(var/i = 0, i < distance, ++i)
-			L.Insert(fromIndex, null)
-			L.Swap(fromIndex, toIndex)
-			L.Cut(toIndex, toIndex + 1)
-	else
-		if(toIndex > fromIndex)
-			var/a = toIndex
-			toIndex = fromIndex
-			fromIndex = a
-
-		for(var/i = 0, i < len, ++i)
-			L.Swap(fromIndex++, toIndex++)
-
-//replaces reverseList ~Carnie
-/proc/reverseRange(list/L, start = 1, end = 0)
-	if(L.len)
-		start = start % L.len
-		end = end % (L.len + 1)
-		if(start <= 0)
-			start += L.len
-		if(end <= 0)
-			end += L.len + 1
-
-		--end
-		while(start < end)
-			L.Swap(start++, end--)
-
-	return L
-
-/proc/counterlist_scale(list/L, scalar)
-	var/list/out = list()
-	for(var/key in L)
-		out[key] = L[key] * scalar
-	. = out
-
-/proc/counterlist_sum(list/L)
-	. = 0
-	for(var/key in L)
-		. += L[key]
-
-/proc/counterlist_normalise(list/L)
-	var/avg = counterlist_sum(L)
-	if(avg != 0)
-		. = counterlist_scale(L, 1 / avg)
-	else
-		. = L
-
-/proc/counterlist_combine(list/L1, list/L2)
-	for(var/key in L2)
-		var/other_value = L2[key]
-		if(key in L1)
-			L1[key] += other_value
-		else
+/*
+ * Holds procs to help with list operations
```

Comment:
did it once AGAIN take an entire file when i did a upstream pull and mark it as the whole thing changed?
WHY
DOES
IT
DO THIS

Reply:
Don't even know what commit did it either..